src/transformers/modeling_utils.py

```diff
@@ -1196,8 +1202,11 @@ def _from_config(cls, config, **kwargs):
         if torch_dtype is not None:
             dtype_orig = cls._set_default_torch_dtype(torch_dtype)
 
-        if use_flash_attention_2:
-            config = cls._check_and_enable_flash_attn_2(config, torch_dtype)
+        config = copy.deepcopy(config)  # We do not want to modify the config inplace in _from_config.
+        config._attn_implementation = kwargs.pop("attn_implementation", None)
```

Comment:
This overrides existing `_attn_implementation` value inside config. And sets it to None when `attn_implementation` is not passed in kwargs ...

Reply:
This is intended, users should not use `_attn_implementation`. Is there a case where you have no choice but to use it?

Reply:
Hi @fxmarty thanks for reply. For context it is the same background as this PR https://github.com/huggingface/transformers/pull/28823 where I tried to unblock export in our benchmark pipeline. 

I guess we misunderstood the error message, and tried to pass `attn_implementation="eager"` to config constructor instead of `from_config` call. 

Regarding your comment though I'm not sure if that is the right behavior. `attn_implementation` is indeed documented in `PretrainedConfig`, and it is not respected if called in this way.